File: hashword-perl/hashword.pl
#! /usr/bin/perl

######################################################################################
# hashword written by: 									
#	Joshua "MrSchism" Embrey [mrschism@sdf.org]					
#	Joseph "Arcarna" Preston [jpreston86@gmail.com]					
# Intial commit: October 9, 2013 							
# Current version: January 22, 2015							
######################################################################################

# Declare hashing digests
use Digest::MD5 qw(md5_hex);
use Digest::SHA qw(sha256_hex);
use HTML::Entities;

# Declare 'my' variables as needed
my $in;

# Correctly clear screen based on OS.
if (($^O eq "MSWin32") or ($^O eq "dos")) {
    system ("cls");
}
else {
    system ("clear");
}

print "\n";
print "Hashword generator v 2.0.3 (build 20150122) \n";  # Identifies version/build info (build info in yyyymmdd format)
print "\n";
print "Please enter your hashword seed.\n";
print "*NOTE: hashword seeds are case sensitive.* \n";
print "\n";
print "Seed: "; # Prompt for initial password seed

# Strip terminal echo
if (($^O eq "MSWin32") or ($^O eq "dos")) {
    system ('@echo off');
}
else {
    system ("stty -echo");
}

$seed = <>;
$seed = HTML::Entities::encode($seed);
chomp $seed;

# Return terminal echo
if (($^O eq "MSWin32") or ($^O eq "dos")) {
    system ('@echo on');
}
else {
    system ("stty echo");
}
print "\n";

$long = sha256_hex($seed); # Declare that long uses sha256 on the seed
$short = md5_hex($seed); # Declare that short uses md5 on the seed

until (defined $in) {
    print "Use Long mode? (yes/NO): ";
    $_ = <>;
    $in = 1 if /^Y/i;
    $in = 0 if /^N/i;
    print "\n";
}
if ($in == 1) {
    $hash = $long;
}
else {
    $hash = $short;
}

print "Your hashword is:\n";
print "\#", $hash, "\n";
print "\n";

$seed = '';
$long = '';
$short = '';
$in = '';
$hash = '';

File: hashword-web/hashword.php
<!DOCTYPE html>
<head>
<title>Hash Your Password</title>
</head>

<body>
<!--
-######################################################################################
-# hashword written by: 									
-#	Joshua "MrSchism" Embrey [mrschism@sdf.org]					
-#	Joseph "Arcarna" Preston [jpreston86@gmail.com]					
-# Intial commit: October 30, 2013 							
-# Current version: November 20, 2013							
-######################################################################################
-->

<?php
$seed = (isset($_POST['password']) ? $_POST['password'] : null); //Takes the user input and applies it to the variable "seed".
$hashtype = (isset($_POST['hashtype']) ? $_POST['hashtype'] : null); //Sets the hashtype based on the radio button input.
?>
<h1>HASH YOUR PASSWORD</h1>
<p>The benefit of hashing your password is improved complexity and resistance to brute-force dictionary attacks.</p>
Please enter your hashword seed.<br />
*NOTE: hashword seeds are case sensitive.*
<hr />
<form action="?" method="POST">
Seed:<input type="password" name="password"><br />
<input type="radio" value="short" name="hashtype" />Short<br />
<input type="radio" value="long" name="hashtype" />Long<br />
<input type="submit" value="Generate Hash" />
</form>
<br />
<br />
<?php
function scrub($seed) {
	$seed = trim($seed);
	$seed = stripslashes($seed);
	$seed = htmlspecialchars($seed);
  	return $seed;
		}

switch ($hashtype) {
		case "short":
			echo "#" , openssl_digest($seed, 'sha256'); //Outputs seed in SHA256 hash.
			break;
		case "long":
			echo "#" , openssl_digest($seed, 'sha512'); //Outputs seed in SHA512 hash.
			break;
		default:
			echo 'Please select Short or Long'; //Outputs when no radio button is selected.
		}
?>
</body>
</html>
